perfetto-sys/src/event.rs

```diff
@@ -187,4 +194,24 @@ impl Drop for TraceEvent {
 
         unsafe { destroy_event(self.category.as_ref().map(|s| s.as_ptr()).unwrap_or(null()), track_id) };
     }
-}
\ No newline at end of file
+}
+
+/// A one-shot instant event: consumes your `EventData`, emits it, then returns.
+pub struct InstantEvent;
```

Comment:
We don't need a structure for this, just a function e.g. `create_instant_event`.

We have a structure for the `TraceEvent` to automatically close the corresponding span on drop, while this is not actual for this type

Reply:
ok, let me change that. FYI: I initially had a function but then I modified it to struct so this logic looked similarly.